src/osd/OSD.cc

```diff
@@ -5864,7 +5864,6 @@ bool OSD::ms_dispatch(Message *m)
 
   do_waiters();
   _dispatch(m);
-  do_waiters();
```

Comment:
@majianpeng probably we can have an assert here to make sure that the `finished` queue is empty?